src/api/spec/controllers/webui/project_controller_spec.rb

```diff
@@ -976,4 +976,91 @@
       it { expect(response).to redirect_to(project_show_path(apache_maintenance_incident_project)) }
     end
   end
+
+  describe 'POST #move_path' do
+    context 'valid project' do
+      let(:repository) { create(:repository, project: apache_project) }
+      let(:path_element) { create(:path_element, repository: repository) }
+
+      context 'without direction' do
+        it { expect { post :move_path, params: { project: apache_project } }.to raise_error ActionController::ParameterMissing }
+      end
+
+      context 'only one path_element' do
+        let(:position) { path_element.position}
+
+        context 'direction up' do
+          before do
+            post :move_path, params: { project: apache_project, repository: repository.id, direction: 'up', path: path_element }
+          end
+
+          it { expect(path_element.reload.position).to eq(position) }
```

Comment:
Looks like the case with no previous/sequent repo is also covered there:

https://github.com/openSUSE/open-build-service/pull/2520/commits/aafc83485b21a47b9293b98703ddef0eb3b98b78#diff-603a086da29f0e3cdbaa1d2dfc369856R1035

https://github.com/openSUSE/open-build-service/pull/2520/commits/aafc83485b21a47b9293b98703ddef0eb3b98b78#diff-603a086da29f0e3cdbaa1d2dfc369856R1054